AxonDeepSeg/visualization/visualize.py

```diff
@@ -1,174 +1,218 @@
+        fig = Figure()
+        FigureCanvas(fig)
+
+        # Drawing the evolution curves
+        ax1 = fig.subplots()
+        ax2 = ax1.twinx()
+        ax1.plot(
+            data_evolution["steps"][start_for_viz:],
+            data_evolution["accuracy"][start_for_viz:],
+            "-",
+            label="accuracy",
+        )
+        ax1.set_ylim(ymin=0)
+        ax2.plot(
+            data_evolution["steps"][start_for_viz:],
+            data_evolution["loss"][start_for_viz:],
+            "-r",
+            label="loss",
+        )
+
+        # Annotating the graph
+        ax1.set_title("Accuracy and loss evolution")
+        ax1.set_xlabel("Epoch")
+        ax1.set_ylabel("Accuracy")
+        ax2.set_ylabel("Loss")
+        return fig
 
-    fig = plt.figure(1)
-    # Drawing the evolution curves
-
-    ax = fig.add_subplot(111)
-    ax.plot(evolution['steps'][start_visu:], evolution['accuracy'][start_visu:], '-', label='accuracy')
-    plt.ylabel('Accuracy')
-    plt.ylim(ymin=0)
-    # plt.ylim(ymax=100)
-
-    ax2 = ax.twinx()
-    ax2.plot(evolution['steps'][start_visu:], evolution['loss'][start_visu:], '-r', label='loss')
-
-    # Annotating the graph
-
-    plt.title('Accuracy and loss evolution')
-    plt.ylabel('Loss')
-    plt.xlabel('Epoch')
-    plt.show()
+    evolution = retrieve_training_data(path_model)
+    fig = _create_figure_helper(evolution)
 
-    return evolution
+    return fig, evolution
 
 
 def visualize_segmentation(path):
     """
-    :param path: path of the folder including the data and the results obtained after by the segmentation process.
-    :return: no return
-    if there is a mask (ground truth) in the folder, scores are calculated : sensitivity, errors and dice
+    :param path: path of the folder including the data and the results obtained
+        after by the segmentation process.
+    :return: list of matplotlib.figure.Figure
+    if there is a mask (ground truth) in the folder, 
+    scores are calculated: sensitivity, errors and dice
     figure(1) segmentation without mrf
     figure(2) segmentation with mrf
-    if there is MyelinSeg.jpg in the folder, myelin and image, myelin and axon segmentated, myelin and groundtruth are represented
+    if there is MyelinSeg.jpg in the folder, myelin and image, myelin and axon
+    segmentated, myelin and groundtruth are represented
     """
-
-    path_img = path + '/image.png'
+    figs = []
+
+    def _create_fig_helper(overlayed_img, fig_title):
+        """
+        Helper function to create a figure
+        :param overlayed_img: the image to add on top on image_init
+        :param fig_title: str title of the figure
+        :return: matplotlib.figure.Figure
+        """
+        fig = Figure()
+        FigureCanvas(fig)
+        ax = fig.subplots()
+        ax.set_title(fig_title)
+        ax.imshow(image_init, cmap="gray")
+        ax.imshow(overlayed_img, cmap="hsv", alpha=0.7)
+        return fig
+
+    path_img = path + "/image.png"
```

Comment:
This line might cause a bug in Windows; we might be better off using `path_img = os.path.join(path, "image.png")` or some other OS-agnostic syntax.

Reply:
Actually nevermind, this should be dealt in another issue.  Github told me that it's a new line, but it's just because you changed the single quotes to double quotes. There's too many of these instances in this file to resolve them right now and permanently.

Reply:
You're right it will be handled in #154. I'm working on that right now!